src/ASM/ASMs2D.C

```diff
@@ -3068,3 +3068,25 @@ size_t ASMs2D::getNoProjectionNodes () const
 {
   return proj->numCoefs_u() * proj->numCoefs_v();
 }
+
+
+void ASMs2D::getNeighbours (NeighArray& neigh) const
+{
+  int N1, N2, dummy;
+  this->getNoStructElms(N1, N2, dummy);
+  for (int i = 0; i < static_cast<int>(this->getNoElms(true)); ++i) {
+    int idx = this->getElmID(i+1)-1;
+    if (idx == -1)
+      continue;
+
+    neigh[idx].resize(4,-1);
+    if (i % N1 > 0)
+      neigh[idx][0] = idx-1;
```

Comment:
I guess this will always work today, but you here implicitly assume that the global element numbers (in `MLGE`) have the same sequence as the patch-wise ordering). But if that is changed this will fail. It would be more consistent with
```
   neigh[idx][0] = this->getElmID(i);
   neigh[idx][1] = this->getElmID(i+2);
   neigh[idx][2] = this->getElmID(i-N1+1);
   neigh[idx][3] = this->getElmID(i+N1+1);
```
or use MLGE lookup directly instead of using getElmID.

Reply:
indeed.